gobblin-data-management/src/test/java/org/apache/gobblin/runtime/embedded/EmbeddedGobblinDistcpTest.java

```diff
@@ -83,6 +118,87 @@ public void test() throws Exception {
     Assert.assertTrue(new File(tmpTarget, fileName).exists());
   }
 
+  @Test
+  public void hiveTest() throws Exception {
+    // Hive Environment set-up
+    HiveConf hiveConf = new HiveConf();
+
+    try {
+      // Start a Hive session in this thread and register the UDF
+      SessionState.start(hiveConf);
+      SessionState.get().initTxnMgr(hiveConf);
+    } catch (HiveException he) {
+      throw new RuntimeException("Failed to start Hive session.", he);
+    }
+    try {
+      jdbcConnector = HiveJdbcConnector.newEmbeddedConnector(2);
+    } catch (SQLException se) {
+      throw new RuntimeException("Cannot initialize the jdbc-connector due to: ", se);
+    }
```

Comment:
Similar question, combine them into one try?

Reply:
Would you help me understand why we need a separate session here instead of using the session created in the setup? Maybe add some comments?

Reply:
Thanks for catching, this is the duplicated code.